nncf/common/pruning/export_helpers.py

```diff
@@ -50,3 +61,212 @@ def get_all_op_aliases(cls):
+
+        if is_grouped_conv(node):
+            output_mask = None
+            if is_depthwise_conv(node):
+                output_mask = input_masks[0]
+
+        node.data['output_mask'] = output_mask
+
+
+class OpTransposeConvolution(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        accept_pruned_input = True
+        if is_grouped_conv(node):
+            if not is_depthwise_conv(node):
+                accept_pruned_input = False
+        return accept_pruned_input
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        input_masks = get_input_masks(node, graph)
+        output_mask = node.data.get('output_mask', None)
+
+        # In case of group convs we can't prune by output filters
+        if is_grouped_conv(node):
+            output_mask = None
+            if is_depthwise_conv(node):
+                output_mask = input_masks[0]
+
+        node.data['output_mask'] = output_mask
+
+
+class OpBatchNorm(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        identity_mask_propagation(node, graph)
+
+
+class OpGroupNorm(DefaultMetaOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        # For Instance Normalization
+        return isinstance(node.layer_attributes, GroupNormLayerAttributes) \
+               and node.layer_attributes.num_groups == node.layer_attributes.num_channels
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph):
+        identity_mask_propagation(node, graph)
+
+
+class OpConcat(DefaultMetaOp):
+    ConvolutionOp = None # type: OpConvolution
+    StopMaskForwardOp = None # type: OpStopMaskForwardOps
+    InputOp = None # type: OpInput
+
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def check_concat(cls, node: NNCFNode, graph: NNCFGraph) -> bool:
+        """
+        Return whether all input sources of node is convolutions or not.
+
+        :param node: Node to determine it's sources
+        :param graph: NNCF graph to work with
+        :return: True if all input sources of node is convolutions
+        """
+
+        for input_node in graph.get_previous_nodes(node):
+            # If input has mask ->  it went from convolution (source of this node is a convolution)
+            if input_node.data.get('output_mask', None) is not None:
+                continue
+
+            source_nodes = get_sources_of_node(input_node, graph, cls.ConvolutionOp.get_all_op_aliases() +
+                                               cls.StopMaskForwardOp.get_all_op_aliases() +
+                                               cls.InputOp.get_all_op_aliases())
+            sources_types = [node.node_type for node in source_nodes] + [input_node.node_type]
```

Comment:
Please be aware of two changes here:

```
if input_node.data.get('output_mask', None) is not None:
```
instead of 

```
if input_node.data.get('output_mask', None) is None:
```

and 

```
sources_types = [node.node_type for node in source_nodes] + [input_node.node_type]
```
instead of 
```
sources_types = [node.node_type for node in source_nodes]
```

Reply:
@mkaglins

Reply:
@daniil-lyakhov Please, describe how the second change (of sources_types) will impact how the algorithm works on known models with Concat.

And, please, test these changes carefully on these models.